app/controllers/incomeSources/add/AddIncomeSourceStartDateController.scala

```diff
@@ -183,69 +146,62 @@ class AddIncomeSourceStartDateController @Inject()(authenticate: AuthenticationP
-    case ex: Exception =>
-      Logger("application").error(s"[AddIncomeSourceStartDateController][handleSubmitRequest] - Error: ${ex.getMessage}")
-      if (isAgent) itvcErrorHandlerAgent.showInternalServerError()
-      else itvcErrorHandler.showInternalServerError()
   }
 
-  private def getCalls(isAgent: Boolean, incomeSourceType: IncomeSourceType): (Call, Call, Call) = {
+  private def getFilledForm(form: Form[DateFormElement],
+                            startDateKey: Option[String],
+                            isUpdate: Boolean): Form[DateFormElement] = {
 
-    (isAgent, incomeSourceType) match {
-      case (false, SelfEmployment) =>
-        (
-          controllers.incomeSources.add.routes.AddBusinessNameController.show(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitSoleTraderBusiness,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showSoleTraderBusiness
-        )
-      case (false, UkProperty) =>
-        (
-          controllers.incomeSources.add.routes.AddIncomeSourceController.show(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitUKProperty,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showUKProperty
-        )
-      case (false, ForeignProperty) =>
-        (
-          controllers.incomeSources.add.routes.AddIncomeSourceController.show(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitForeignProperty,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showForeignProperty
-        )
-      case (true, SelfEmployment) =>
-        (
-          controllers.incomeSources.add.routes.AddBusinessNameController.showAgent(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitSoleTraderBusinessAgent,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showSoleTraderBusinessAgent
-        )
-      case (true, UkProperty) =>
-        (
-          controllers.incomeSources.add.routes.AddIncomeSourceController.showAgent(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitUKPropertyAgent,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showUKPropertyAgent
-        )
-      case (true, ForeignProperty) =>
-        (
-          controllers.incomeSources.add.routes.AddIncomeSourceController.showAgent(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitForeignPropertyAgent,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showForeignPropertyAgent
+    (startDateKey, isUpdate) match {
+      case (Some(key), true) if Try(LocalDate.parse(key)).toOption.isDefined =>
+        form.fill(
+          DateFormElement(
+            LocalDate.parse(key)
+          )
```

Comment:
* we would need to log error in case we are not able to parse String => Date ?
* you can try something like that
`    startDateKey.flatMap(key => 
      if (isUpdate){
        Try(LocalDate.parse(key)).toOption.map( date =>
          form.fill(DateFormElement(date))
        )
      } else {
        None
      }
    ).getOrElse{
      // log error here
      form
    }`